Packet++/src/Asn1Codec.cpp

```diff
@@ -0,0 +1,569 @@
+			m_EncodedValue = nullptr;
+		}
+	}
+
+	Asn1GenericRecord::Asn1GenericRecord(Asn1TagClass tagClass, bool isConstructed, uint8_t tagType, const uint8_t* value, size_t valueLen)
+	{
+		m_TagType = tagType;
+		m_TagClass = tagClass;
+		m_IsConstructed = isConstructed;
+		m_Value = new uint8_t[valueLen];
+		m_FreeValueOnDestruction = true;
+		memcpy(m_Value, value, valueLen);
+		m_ValueLength = valueLen;
+		m_TotalLength = m_ValueLength + 2;
+	}
+
+	Asn1GenericRecord::~Asn1GenericRecord()
+	{
+		if (m_Value && m_FreeValueOnDestruction)
+		{
+			delete m_Value;
+		}
+	}
+
+	void Asn1GenericRecord::decodeValue(uint8_t* data, bool lazy)
+	{
+		m_Value = data;
+	}
+
+	std::vector<uint8_t> Asn1GenericRecord::encodeValue() const
+	{
+		return {m_Value, m_Value + m_ValueLength};
+	}
+
+	Asn1ConstructedRecord::Asn1ConstructedRecord(Asn1TagClass tagClass, uint8_t tagType, const std::vector<Asn1Record*>& subRecords)
+	{
+		m_TagType = tagType;
+		m_TagClass = tagClass;
+		m_IsConstructed = true;
+
+		size_t recordValueLength = 0;
+		for (auto record : subRecords)
+		{
+			auto encodedRecord = record->encode();
+			auto copyRecord = Asn1Record::decode(encodedRecord.data(), encodedRecord.size(), false);
+			m_SubRecords.pushBack(copyRecord.release());
+			recordValueLength += encodedRecord.size();
+		}
+
+		m_ValueLength = recordValueLength;
+		m_TotalLength = recordValueLength + 2;
+	}
+
+	void Asn1ConstructedRecord::decodeValue(uint8_t* data, bool lazy)
+	{
+		if (!(data || m_ValueLength))
+		{
+			return;
+		}
+
+		auto value = data;
+		auto valueLen = m_ValueLength;
+
+		while (valueLen > 0)
+		{
+			auto subRecord = Asn1Record::decodeInternal(value, valueLen, lazy);
+			value += subRecord->getTotalLength();
+			valueLen -= subRecord->getTotalLength();
+
+			m_SubRecords.pushBack(subRecord);
+		}
+	}
+
+	std::vector<uint8_t> Asn1ConstructedRecord::encodeValue() const
+	{
+		std::vector<uint8_t> result;
+		result.reserve(m_ValueLength);
+
+		for (auto record : m_SubRecords)
+		{
+			auto encodedRecord = record->encode();
+			result.insert(result.end(), std::make_move_iterator(encodedRecord.begin()), std::make_move_iterator(encodedRecord.end()));
+		}
+		return result;
+	}
+
+	Asn1SequenceRecord::Asn1SequenceRecord(const std::vector<Asn1Record*>& subRecords)
+		: Asn1ConstructedRecord(Asn1TagClass::Universal, static_cast<uint8_t>(Asn1UniversalTagType::Sequence), subRecords)
+	{}
+
+	Asn1SetRecord::Asn1SetRecord(const std::vector<Asn1Record*>& subRecords)
+			: Asn1ConstructedRecord(Asn1TagClass::Universal, static_cast<uint8_t>(Asn1UniversalTagType::Set), subRecords)
+	{}
+
+	Asn1PrimitiveRecord::Asn1PrimitiveRecord(uint8_t tagType) : Asn1Record()
+	{
+		m_TagType = tagType;
+		m_TagClass = Asn1TagClass::Universal;
+		m_IsConstructed = false;
+	}
+
+	Asn1IntegerRecord::Asn1IntegerRecord(uint32_t value) : Asn1PrimitiveRecord(static_cast<uint8_t>(Asn1UniversalTagType::Integer))
+	{
+		m_Value = value;
+
+		if (m_Value <= std::pow(2, sizeof(uint8_t) * 8))
```

Comment:
we can use `std::numeric_limits<uint8_t>::max();`

Reply:
Fixed in this commit: https://github.com/seladb/PcapPlusPlus/pull/1353/commits/4b4865f2aa01e3c5ac619f91d7a6d0ff0227934c